packages/core/src/browser/preferences/preference-provider.ts

```diff
@@ -14,16 +14,24 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 
-// tslint:disable:no-any
-
 import { injectable } from 'inversify';
 import { Disposable, DisposableCollection, Emitter, Event } from '../../common';
 import { Deferred } from '../../common/promise-util';
+import { PreferenceScope, PreferenceChange } from './preference-service';
+
+export namespace PreferenceProviderPriority {
+    export const NA = -1;
+    export const Default = 0;
+    export const User = 1;
+    export const Workspace = 2;
+    export const Folder = 3;
+}
 
 @injectable()
-export class PreferenceProvider implements Disposable {
-    protected readonly onDidPreferencesChangedEmitter = new Emitter<void>();
-    readonly onDidPreferencesChanged: Event<void> = this.onDidPreferencesChangedEmitter.event;
+export abstract class PreferenceProvider implements Disposable {
+
+    protected readonly onDidPreferencesChangedEmitter = new Emitter<PreferenceChange>();
+    readonly onDidPreferencesChanged: Event<PreferenceChange> = this.onDidPreferencesChangedEmitter.event;
```

Comment:
it is breaking change, it used to be composite event, to indicate that there are some preferences changed, not one like it is now

Reply:
yes i changed the data that is emitted with the Event, but i am not sure why it is a breaking change:

- PreferenceProvider is used internally. Clients only use PreferenceService.
- Even if there are clients listening to `PreferenceProvider.onDidPreferencesChanged`, the clients' code is not broken by this change, because the clients' code does not expect getting anything from the event.

Reply:
`MonacoConfiguration` should completely reconcile all preferences on a change:  https://github.com/theia-ide/theia/blob/6a3444d341b4e16048a18f3588d0249f33d8f57c/packages/monaco/src/browser/monaco-configurations.ts#L43

Before it was not for each preference name, but for composite. You changed make it very expensive now, instead of reconciling once, it reconciles for each preference name.

Please put back `onPreferencesChangedEmitter` or rewrite `MonacoConfiguration` more efficient without breaking it.